nemo/collections/common/callbacks/callbacks.py

```diff
@@ -30,3 +32,40 @@ def on_epoch_end(self, trainer, pl_module):
         curr_time = time.time()
         duration = curr_time - self.epoch_start
         trainer.logger.log_metrics({"epoch_time": duration}, step=trainer.global_step)
+
+
+class LogTrainValidLossCallback(Callback):
+    @rank_zero_only
+    def on_train_start(self, trainer, pl_module):
+        logging.info("Training started")
+
+    @rank_zero_only
+    def on_train_batch_end(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
+        print_freq = trainer.row_log_interval
```

Comment:
This will print every single batch (since PTL default is 10, nemo default is 1, not 1.0).

Reply:
yes, its provided by user based on what % of num_batches or exact number he/she requires.